### ISSUE TITLE
examples/propagation1d.c includes laik-internal.h

### ISSUE THREAD

**Issue by AlexanderKurtz** [envelope-project/laik#111]
Disclaimer: This came up in PR #99, but it really should be a dedicated issue.

```examples/propagation1d.c``` includes ```laik-internal.h``` but it probably shouldn't. Instead, liblaik should gain proper getters and setters so that propagation1d.c can be changed to only include ```laik.h```.

CC: @amir-raoofy @weidendo

**Comment by weidendo** [comment]
This is the partitioner accessing structs directly. I think we should have accessors already in place?

**Comment by AlexanderKurtz** [comment]
Really? I didn't find these getters/setters when I originally proposed #99. Have they been added in the mean time?

For the record, here's the full IWYU-provided include list for ```propagation1d.c```:

```C
#include <assert.h>         // for assert
#include <laik-internal.h>  // for _Laik_Partitioning, Laik_TaskSlice_Gen
#include <laik.h>           // for Laik_Partitioning, Laik_Space, laik_map_def1
#include <stdint.h>         // for int64_t, uint64_t
```

**Comment by amir-raoofy** [comment]
Hi Alexander, 

I wrote the propagation1d when we still did not have the API. But then I continued working on 2d version which uses the later developed API by Josef. At the moment I am a bit busy for adapting the code. In fact, this code is not so important since the 2d version already is more developed and the unit tests for the propagation2d should be already enough for checking the functionalities that I need for the LULESH. If it is not urgent, I will adapt it later and if it is, I think it will not bother if we remove 1d version.

**Comment by weidendo** [comment]
Writing a partitioner is a typical thing to do, also using an already exisiting partitioning as input in a partitioner. The LULESH port does this, and this cannot include laik-internal.h as external project.

See runLuleshNodePartitioner() in propagation2d.c:

    Laik_Space* baseSpace = laik_partitioning_get_space(baseP);
    ....
    assert(laik_space_getdimensions(space) ==
           laik_space_getdimensions(baseSpace));

    int sliccountE = laik_partitioning_slicecount(baseP);
    for(int i = 0; i < sliccountE; i++) {
        Laik_TaskSlice* ts = laik_partitioning_get_tslice(baseP, i);
        const Laik_Slice* s = laik_taskslice_get_slice(ts);
        int task = laik_taskslice_get_task(ts);
        ...

That should be enough for propagation1d.c

**Comment by amir-raoofy** [comment]
Yes and the API is there.

**Comment by weidendo** [comment]
Was fixed in ffb4b3d4
